tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py

```diff
@@ -13,16 +13,31 @@
 # limitations under the License.
 import contextlib
 import logging
+import os
 import pathlib
 from typing import Optional
+import time
+import threading
 
 import mako.template
+import tenacity
 import yaml
 
 from framework.infrastructure import k8s
+from kubernetes import client, utils, watch
+import kubernetes.client.rest
 
 logger = logging.getLogger(__name__)
 
+if "KOKORO_ARTIFACTS_DIR" in os.environ:
+    _TEST_LOG_BASE_DIR = os.environ["KOKORO_ARTIFACTS_DIR"]
```

Comment:
I don't think this framework should be decoupled from kokoro and contain no kokoro-specific logic.
Let's make this an argument. We can call it --output-dir, and make it so other modules also can save output to it.
Similar to https://abseil.io/docs/python/guides/flags#using-flags-for-global-variablesconstants